CmswebApI/Controllers/CoursesController.cs

```diff
@@ -47,67 +48,46 @@ public async Task<ActionResult<IEnumerable<CourseDto>>> GetCoursesAsync()
             try
             {
                 IEnumerable<Course> courses = await _cmsrepository.GetAllCoursesAsync();
-                var result = MapCourseToCourseDto(courses);
+                var result = MappingHelper.MapCourseModelListToCourseDtoList(courses);
                 return result.ToList();
             }
             catch (System.Exception ex)
             {
                 return StatusCode(StatusCodes.Status500InternalServerError, ex.Message);
             }
         }
-        //Custom mapper functions
-        private CourseDto MapCourseToCourseDto(Course course)
-        {
-            return new CourseDto()
-            {
-                CourseID = course.CourseID,
-                CourseName = course.CourseName,
-                CourseDuration = course.CourseDuration,
-                CourseType = (CmswebApI.DTOs.COURSE_TYPE)course.CourseType
-            };
-        }
-        private IEnumerable<CourseDto> MapCourseToCourseDto(IEnumerable<Course> courses)
-        {
-            IEnumerable<CourseDto> result;
-
-            result = courses.Select(c => new CourseDto()
-            {
-                CourseID = c.CourseID,
-                CourseName = c.CourseName,
-                CourseDuration = c.CourseDuration,
-                CourseType = (CmswebApI.DTOs.COURSE_TYPE)c.CourseType
-            });
-            return result;
-        }
         [HttpPost]
         public ActionResult<CourseDto> AddCourse([FromBody] CourseDto courseDto)
         {
             try
             {
-                var courseModel =  MapCourseDtoToCourse(courseDto);
-
-                var newCourse =  _cmsrepository.AddCourse(courseModel);
-
-                return MapCourseToCourseDto(newCourse);
+                var courseModel = MappingHelper.MapCourseDtoToCourse(courseDto);
+                var newCourse = _cmsrepository.AddCourse(courseModel);
+                return MappingHelper.MapCourseModelToCourseDto(newCourse);
             }
             catch (System.Exception ex)
             {
                 return StatusCode(StatusCodes.Status500InternalServerError, ex.Message);
-
             }
-
         }
 
-        
-        private Course MapCourseDtoToCourse(CourseDto courseDto)
+        [HttpGet("{courseID}")]
+        public ActionResult<CourseDto> GetCourse(int courseID)
```

Comment:
Make the route async, and also rename it to GetCourseById for better readability.